### PR TITLE
[ISSUE-709] Improve release workflow

### PR DESCRIPTION
**Pull request by misanthropicat**
### Issue dell/csi-baremetal#709

Added workflow input parameter for branch to release from.

## PR checklist
- [ ] Add link to the issue
- [x] Choose Project
- [x] Choose PR label
- [ ] New unit tests added
- [ ] Modified code has meaningful comments
- [ ] All TODOs are linked with the issues
- [ ] All comments are resolved

## Testing
It's impossible to test in this repo before this PR is merged as will be executed workflow code from master branch of this repo even if another branch as code source will be chosen in csi-baremetal repo. See [this line comment](https://github.com/dell/csi-baremetal/blob/d949b44f41506a4a173d6039b196f4d732e8d668/.github/workflows/release.yml#L40)
Tested on fork. Releases created by automation:
https://github.com/misanthropicat/csi-baremetal/releases/tag/v1.0.0-test
https://github.com/misanthropicat/csi-baremetal-operator/releases/tag/v1.0.0-test

### CODE REVIEW

**Review by rkiselenko — APPROVED** [review]

**Review by safronovD — APPROVED** [review]

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -1,151 +1,157 @@
...
-
-      # - name: Send slack notification
-      #   uses: rtCamp/action-slack-notify@v2
-      #   env:
-      #     SLACK_WEBHOOK: ${{ secrets.SLACK_WEBHOOK }}
-      #     SLACK_CHANNEL: test-private-channel
-      #     SLACK_COLOR: ${{ job.status }}
-      #     SLACK_TITLE: 'Release ${{ github.event.inputs.release_tag }} is ready for review'
-      #     SLACK_MESSAGE: 'Please review this PR: ${{ steps.pr.outputs.pull-request-url }}'
+name: Release Charts
```
Comment (safronovD) [inline .github/workflows/release.yml:1]: Why diff shows that the whole file was changed?

Comment (misanthropicat) [reply .github/workflows/release.yml:1]: Idk, have the same question.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -1,151 +1,157 @@
-name: Release Charts
```
Comment (mishoyama) [inline .github/workflows/release.yml:1]: Looks like something wrong the formatting. Hard to review.

**Review by mishoyama — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml (hunk 2 above)
Comment (misanthropicat) [reply .github/workflows/release.yml]: Fixed. Please take a look again.

**Comment by mishoyama** [comment]
@misanthropicat please create temporary release for test purposes

**Comment by misanthropicat** [comment]
> @misanthropicat please create temporary release for test purposes

I did it on fork + it's impossible to do before this PR merge as will be executed workflow code from master branch of this repo even if another branch as code source will be chosen. See [this line comment](https://github.com/dell/csi-baremetal/blob/d949b44f41506a4a173d6039b196f4d732e8d668/.github/workflows/release.yml#L40)

**Comment by mishoyama** [comment]
> > @misanthropicat please create temporary release for test purposes
> 
> I did it on fork + it's impossible to do before this PR merge as will be executed workflow code from master branch of this repo even if another branch as code source will be chosen. See [this line comment](https://github.com/dell/csi-baremetal/blob/d949b44f41506a4a173d6039b196f4d732e8d668/.github/workflows/release.yml#L40) Test releases on fork: https://github.com/misanthropicat/csi-baremetal/releases/tag/v1.0.0-test, https://github.com/misanthropicat/csi-baremetal-operator/releases/tag/v1.0.0-test

@misanthropicat please add this information to the **Testing** section

**Comment by misanthropicat** [comment]
> > > @misanthropicat please create temporary release for test purposes
> > 
> > 
> > I did it on fork + it's impossible to do before this PR merge as will be executed workflow code from master branch of this repo even if another branch as code source will be chosen. See [this line comment](https://github.com/dell/csi-baremetal/blob/d949b44f41506a4a173d6039b196f4d732e8d668/.github/workflows/release.yml#L40) Test releases on fork: https://github.com/misanthropicat/csi-baremetal/releases/tag/v1.0.0-test, https://github.com/misanthropicat/csi-baremetal-operator/releases/tag/v1.0.0-test
> 
> @misanthropicat please add this information to the **Testing** section

Done.

**Review by mishoyama — APPROVED** [review]

**Comment by mishoyama** [comment]
@misanthropicat do we need to update documentation?

**Pull request merged**